### PR TITLE
Default `noDeploy` to an empty list

### LINKED ISSUE #304: Remove boto3 and botocore in noDeploy to adhere to AWS best practices.
Recently had a discussion with AWS employees about dependency handling and their recommendation is to bundle all your own packages, including _boto3_ and _botocore_. This is not the default behavior of serverless-python-requirements unfortunately. I've opened this issue to discuss removing those two packages or even rethink the _noDeplopy_ option so it better aligns with AWS recommended practices.

### PR DESCRIPTION
**Pull request by jpadilla**
Closes #304 

Based on https://github.com/UnitedIncome/serverless-python-requirements/issues/304#issuecomment-454787377 I changed the default for `noDeploy` to an empty array. Not entirely sure if it might have been enough to remove `boto3` and `botocore` instead.

### CODE REVIEW

**Comment by dschep** [comment]
Could you try adding `- cmd: pip install -U pip` before this line https://github.com/UnitedIncome/serverless-python-requirements/blob/master/appveyor.yml#L3

**Comment by jpadilla** [comment]
@dschep thanks! I made tests pass on appveyor, but obviously broken now on circleci...

File: circle.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ jobs:
           sudo ./install.sh /usr/local
       # other deps
       - run: sudo apt -y update && sudo apt -y install python-pip python2.7 curl unzip
+      # update pip
```
Comment (dschep) [inline circle.yml]: Sorry, only intended for you to update the appveryor config since circle was passing. so remove it here 😉

Comment (jpadilla) [reply circle.yml]: @dschep added this as I was trying to make circleci pass again, I'll remove(Also my bad for amending, I can see how that doesn't also help reviews)

For appveyor to pass I did as you suggest it but also updated this https://github.com/UnitedIncome/serverless-python-requirements/pull/312/files#diff-45fdab74782ebb95f53bf5fb376eeffb

**Comment by jdufresne** [comment]
Hi @jpadilla @dschep 

I'm interested in seeing this improvement land. Let me know if there is anything I can do to lend a helping hand.

**Comment by jdufresne** [comment]
The "Omitting Packages" section in README will need some editing to make it accurate.

**Comment by jpadilla** [comment]
@jdufresne just updated this PR with the latest changes in master. Tests are failing again, so I'll need to take a look. Any help is always appreciated.

**Comment by jdufresne** [comment]
The section in README starting with "To include the default omitted packages ..." should now be removed as well.

File: tests/base/requirements-w-hashes.txt [hunk 2]
```diff
@@ -87,3 +87,5 @@ werkzeug==0.14.1 \
     --hash=sha256:c3fd7a7d41976d9f44db327260e263132466836cef6f91512889ed60ad26557c \
     --hash=sha256:d5da73735293558eb1651ee2fddc4d0dedcfa06538b8813a2e20011583c9e49b \
     # via flask
+futures==3.2.0 \
+    --hash=sha256:ec0a6cb848cc212002b9828c3e34c675e0c9ff6741dc445cab6fdd4e1085d1f1 \
```
Comment (jdufresne) [inline tests/base/requirements-w-hashes.txt]: Why was this added? I looks incorrect to me. I think removing it may allow the tests to pass.

**Comment by jpadilla** [comment]
Appveyor seems to still be failing but seems like for some other reason: https://ci.appveyor.com/project/dschep/serverless-python-requirements/builds/24744154#L186

**Comment by jpadilla** [comment]
@dschep @jdufresne 🙌 thanks!

**Comment by stevenolen** [comment]
👋 any chance a release could be cut to include this? It's pretty clunky to have to continue adding a few (soon-to-be) meaningless lines to every project's serverless.yml.
